packages/map-template/src/components/MapTemplate/MapTemplate.jsx

```diff
@@ -88,6 +86,89 @@ function MapTemplate({ apiKey, gmApiKey, mapboxAccessToken, venue, locationId, p
 
     const [pushAppView, goBack, currentAppView, currentAppViewPayload, appStates] = useAppHistory();
 
+    /**
+     * Ensure that MapsIndoors Web SDK is available.
+     *
+     * @returns {Promise}
+     */
+    function initializeMapsIndoorsSDK() {
+        return new Promise((resolve) => {
+            if (window.mapsindoors !== undefined) {
+                return resolve();
+            }
+
+            const miSdkApiTag = document.createElement('script');
+            miSdkApiTag.setAttribute('type', 'text/javascript');
+            miSdkApiTag.setAttribute('src', 'https://app.mapsindoors.com/mapsindoors/js/sdk/4.23.1/mapsindoors-4.23.1.js.gz');
+            document.body.appendChild(miSdkApiTag);
+            miSdkApiTag.onload = () => resolve();
+        });
+    }
+
+    /**
+     * Wait for the MapsIndoors JS SDK to be initialized, then react on changes on the props.
+     * React on changes in the MapsIndoors API key by fetching the required data.
+     */
+    useEffect(() => {
```

Comment:
Everything inside this will run every time ANY of the props change. That seems unnecessary, and I didn't think this through when we discussed the solution earlier.

Could we instead have a state hook, for example `mapsindoorsSDKAvailable` that is set to true when the onload for the script resolves?

And then you could add that as dependency on the smaller useEffects and only run the original code if the SDK is available.

Reply:
Refactored the `useEffect` and added `mapsindoorsSDKAvailable` state. Should we have a default value for it?

Reply:
I guess the default should be false until we make the check and resolve.

Reply:
Added default value to be false